### PR TITLE
Use responseCause instead of CapturedServiceException

### PR DESCRIPTION
**Pull request by minwoox**

### CODE REVIEW

**Review by jrhee17 — APPROVED** [review]
Thanks for the cleanup @minwoox 👍 🙇 👍

**Review by ikhoon — APPROVED** [review]
Thanks!

**Review by trustin — APPROVED** [review]
Thanks for cleaning it up, @minwoox :bow:

File: core/src/main/java/com/linecorp/armeria/server/AbstractHttpResponseHandler.java [hunk 1]
```diff
@@ -111,7 +112,10 @@ final boolean failIfStreamOrSessionClosed() {
         // 3. After successfully flushing, the listener requests next data and
         //    the subscriber attempts to write the next data to the stream closed at 2).
         if (!isWritable()) {
-            Throwable cause = CapturedServiceException.get(reqCtx);
+            Throwable cause = null;
+            if (reqCtx.log().isAvailable(RequestLogProperty.RESPONSE_CAUSE)) {
+                cause = reqCtx.log().ensureAvailable(RequestLogProperty.RESPONSE_CAUSE).responseCause();
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/AbstractHttpResponseHandler.java:117]: Side note for Kotlin: It looks verbose to use `isAvailable(RequestLogProperty.RESPONSE_CAUSE)` and `ensureAvailable(RequestLogProperty.RESPONSE_CAUSE)` to get the value when it is available.

The following style might be useful to take advantage of Kotlin null [safe calls](https://kotlinlang.org/docs/null-safety.html#safe-calls).
```kt
val cause: Throwable? = reqCtx.log().getIfAvailable(RequestLogProperty.RESPONSE_CAUSE)?.responseCause()
```

**Review by ikhoon — COMMENTED** [review]
Still LGTM

File: core/src/main/java/com/linecorp/armeria/server/AbstractHttpResponseHandler.java (hunk 1 above)
Comment (trustin) [reply core/src/main/java/com/linecorp/armeria/server/AbstractHttpResponseHandler.java:117]: Care to create an issue labled with `good-first-issue`?

Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/server/AbstractHttpResponseHandler.java:117]: Sure!

**Pull request merged**